subprojects/packagefiles/cfitsio/meson.build

```diff
@@ -0,0 +1,270 @@
+project(
+  'cfitsio',
+  'c',
+  version: '4.5.0',
+  meson_version: '>=1.1.0',
+  license: 'CFITSIO',
+  license_files: ['licenses/License.txt'],
+)
+has_fortran = add_languages(
+  'fortran',
+  native: false,
+  required: get_option('fortran'),
+)
+
+SOVERSION = 10
+
+cc = meson.get_compiler('c')
+libm = cc.find_library('m')
+
+if target_machine.system() != 'windows'
+  curl = dependency(
+    'libcurl',
+    required: get_option('curl'),
+    fallback: ['curl', 'libcurl'],
+  )
+  if curl.found()
+    add_project_arguments('-DCFITSIO_HAVE_CURL', language: 'c')
+  endif
+endif
+
+threads = dependency('threads', required: get_option('reentrant'))
+if threads.found()
+  add_project_arguments('-D_REENTRANT', language: 'c')
+endif
+
+zlib = dependency('zlib', fallback: ['zlib', 'zlib_dep'])
+
+if target_machine.system() != 'windows'
+  bzip2 = dependency(
```

Comment:
bzip2 doesn't provide pkg-config files, please detect it with cc.find_library instead.

Reply:
both the platforms i developed this for (arch linux and macos) have pkg-config files for bzip2. but since they are not provided by up stream i will switch to find_library()